src/terminal/message.rs

```diff
@@ -2,59 +2,86 @@ use super::emoji;
 
-fn message(msg: &str) {
-    println!("{}", msg);
-}
+pub trait Message {
+    fn message(msg: &str);
 
-pub fn billboard(msg: &str) {
-    let billboard = Billboard::builder()
-        .border_style(BorderStyle::Round)
-        .border_color(BorderColor::Cyan)
-        .margin(1)
-        .build();
-    billboard.display(msg);
-}
+    fn info(msg: &str) {
+        let msg = format!("{} {}", emoji::INFO, msg);
+        Self::message(&msg);
+    }
 
-pub fn info(msg: &str) {
-    let msg = format!("{} {}", emoji::INFO, msg);
-    message(&msg);
-}
+    fn warn(msg: &str) {
+        let msg = format!("{} {}", emoji::WARN, msg);
+        Self::message(&msg);
+    }
 
-pub fn warn(msg: &str) {
-    let msg = format!("{} {}", emoji::WARN, msg);
-    message(&msg);
-}
+    fn success(msg: &str) {
+        let msg = format!("{} {}", emoji::SPARKLES, msg);
+        Self::message(&msg);
+    }
 
-pub fn success(msg: &str) {
-    let msg = format!("{} {}", emoji::SPARKLES, msg);
-    message(&msg);
-}
+    fn user_error(msg: &str) {
+        let msg = format!("{} {}", emoji::EYES, msg);
+        Self::message(&msg);
+    }
 
-pub fn user_error(msg: &str) {
-    let msg = format!("{} {}", emoji::EYES, msg);
-    message(&msg);
-}
+    fn working(msg: &str) {
+        let msg = format!("{} {}", emoji::SWIRL, msg);
+        Self::message(&msg);
+    }
 
-pub fn working(msg: &str) {
-    let msg = format!("{} {}", emoji::SWIRL, msg);
-    message(&msg);
-}
+    fn preview(msg: &str) {
+        let msg = format!("{} {}", emoji::WORKER, msg);
+        Self::message(&msg);
+    }
 
-pub fn preview(msg: &str) {
-    let msg = format!("{} {}", emoji::WORKER, msg);
-    message(&msg);
+    fn help(msg: &str) {
+        let msg = format!("{} {}", emoji::SLEUTH, msg);
+        Self::message(&msg);
+    }
+
+    fn billboard(msg: &str);
+    fn deprecation_warning(msg: &str);
 }
 
-pub fn help(msg: &str) {
-    let msg = format!("{} {}", emoji::SLEUTH, msg);
-    message(&msg);
+pub struct StdOut;
+
+impl Message for StdOut {
+    fn message(msg: &str) {
+        println!("{}", msg);
+    }
+
+    fn billboard(msg: &str) {
+        let billboard = Billboard::builder()
+            .border_style(BorderStyle::Round)
+            .border_color(BorderColor::Cyan)
+            .margin(1)
+            .build();
+        billboard.display(msg);
+    }
+
+    fn deprecation_warning(msg: &str) {
+        let bb = Billboard::builder()
+            .border_style(BorderStyle::Round)
+            .border_color(BorderColor::Red)
+            .margin(1)
+            .build();
+        bb.display(msg);
+    }
 }
 
-pub fn deprecation_warning(msg: &str) {
-    let bb = Billboard::builder()
-        .border_style(BorderStyle::Round)
-        .border_color(BorderColor::Red)
-        .margin(1)
-        .build();
-    bb.display(msg);
+pub struct StdErr;
+
+impl Message for StdErr {
+    fn message(msg: &str) {
+        eprintln!("{}", msg);
+    }
+
+    fn billboard(_msg: &str) {
```

Comment:
this is unfortunate, esp since these are generally not program output. @EverlastingBugstopper any chance you'd be open to implementing a formatter only billboard? <3

Reply:
Yes most definitely, I've been meaning to put some work into that crate

Reply:
let's chat about this in wrangler contributors- i'd be willing to take a stab at it if you could explain what `formatter only` means here :)